packages/flutter/lib/src/material/material_localizations.dart

```diff
@@ -2,35 +2,53 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
 import 'package:flutter/widgets.dart';
 
-/// Interface for localized resource values for the material widgets.
-///
-/// This class provides a default placeholder implementation that returns
-/// hard-coded American English values.
+import 'i18n/localizations.dart';
+
+/// Localized strings for the material widgets.
 class MaterialLocalizations {
-  /// Create a placeholder object for the localized resources of material widgets
-  /// which only provides American English strings.
-  const MaterialLocalizations();
+  /// Construct an object that defines the material widgets' localized strings
+  /// for the given `locale`.
+  const MaterialLocalizations(this.locale) : assert(locale != null);
 
   /// The locale for which the values of this class's localized resources
   /// have been translated.
-  Locale get locale => const Locale('en', 'US');
+  final Locale locale;
```

Comment:
This class is also the interface that other MaterialLocalizations implementations will implement or extend, so it doesn't really make sense for this to be a field.